pacoloco.go

```diff
@@ -290,16 +313,28 @@ func handleRequest(w http.ResponseWriter, req *http.Request) error {
 			ifLater = stat.ModTime()
 		}
 
-		for _, url := range repo.getUrls() {
-			downloaded, err = downloadFile(url+path+"/"+fileName, filePath, ifLater, w)
+		for _, upsteam_url := range repo.getUrls() {
+			upsteam_url, err := url.Parse(upsteam_url)
+			downloaded, err = downloadFile(upsteam_url.JoinPath(path+"/"+fileName).String(), filePath, ifLater, w)
```

Comment:
it makes sense to keep the original `downloadFile(url+path+"/"+fileName, filePath, ifLater, w)` code and instead use `host` variable variable:
```go
u, err := url.Parse(upsteam_url)
if err != nil { return err }
host := u.Host
...
downloadedFiles.WithLabelValues(repoName, host).Inc()
```

Reply:
I've moved that to the downloadFile function. This way metrics cover prefetch requests with less duplication.